pandas/indexes/base.py

```diff
@@ -1403,46 +1407,31 @@ def _ensure_compat_append(self, other):
             to_concat.append(other)
 
         for obj in to_concat:
-            if (isinstance(obj, Index) and obj.name != name and
-                    obj.name is not None):
-                name = None
-                break
+            if not isinstance(obj, Index):
+                raise TypeError('all inputs must be Index')
```

Comment:
(a bit similar with the `equals` case)

Currently, we are accepting a list or tuple to be added (in case that it is a list of list, in the flat case you get an error):

```
In [56]: idx = pd.Index([1,2])

In [57]: idx.append([[1,2]])
Out[57]: Int64Index([1, 2, 1, 2], dtype='int64')
```

The docstring of append also says for `other`: "Index or list/tuple of indices"